core/src/main/java/org/apache/calcite/avatica/remote/looker/LookerResponseParser.java

```diff
@@ -82,19 +86,23 @@ static Object deserializeValue(JsonParser parser, ColumnMetaData columnMetaData)
       return parser.getShortValue();
     case PRIMITIVE_INT:
     case INTEGER:
-      return parser.getIntValue();
+      return parser.getValueAsInt();
     case PRIMITIVE_LONG:
     case LONG:
-      return parser.getLongValue();
+      return parser.getValueAsLong();
     case PRIMITIVE_FLOAT:
     case FLOAT:
       return parser.getFloatValue();
     case PRIMITIVE_DOUBLE:
     case DOUBLE:
-      return parser.getDoubleValue();
+      return parser.getValueAsDouble();
     case NUMBER:
       // NUMBER is represented as BigDecimal
-      return parser.getDecimalValue();
+      if (parser.currentToken() == JsonToken.VALUE_STRING) {
+        return NumberInput.parseBigDecimal(parser.getValueAsString());
```

Comment:
I don't know that much about `JsonToken`, what's a case where this condition would return `true`?

Reply:
From what I observed, you could have quoted numbers or unquoted numbers being parsed. If they are quoted then the parser will state that the current token is a string, if they were unquoted then it will be something like `VALUE_INT` (maybe `VALUE_NUMBER` in this case).

So if "1.2" was sent over we'll have to do that parseBigDecimal. Kind of annoying API in that way because there are other methods that let you call `getValueAs<Type>` but not for all types?  For example, `getIntValue` vs `getValueAsInt` - the latter will actually delegate to getIntValue if current token is `VALUE_INT` otherwise parse the string to an int. (Anyway, maybe this is just be me not fully understanding all the ways parsing data can be problematic but would have been nice to update all calls from get<Type>Value to getValueAs<Type>)